Source/WebCore/dom/Document.cpp

```diff
@@ -383,6 +384,33 @@ struct Document::PendingScrollEventTargetList {
 
 static const Seconds intersectionObserversInitialUpdateDelay { 2000_ms };
 
+class CallbackForContainIntrinsicSize : public NativeResizeObserverCallback {
+public:
+    void onResize(const Vector<Ref<ResizeObserverEntry>>&) final;
+};
+
+void CallbackForContainIntrinsicSize::onResize(const Vector<Ref<ResizeObserverEntry>>& entries)
+{
+    for (auto& entry : entries) {
+        if (auto* target = entry->target()) {
+            if (!target->isConnected()) {
+                target->document().unobserveForContainIntrinsicSize(*target);
+                continue;
+            }
+
+            if (auto* box = target->renderBox(); !box || box->shouldApplySizeContainment()) {
```

Comment:
BTW, you should check for "skips contents" instead of "has size containment". See the resolution in https://github.com/w3c/csswg-drafts/issues/7807#issuecomment-1317388716